specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/networkData.json

```diff
@@ -0,0 +1,408 @@
+              "properties": {
+                "actualTrafficVolume": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The actual traffic outbound volume"
+                },
+                "connectionsDetails": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/ConnectionDetails"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "unusedPorts": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The unused ports details",
+          "properties": {
+            "inboundConnectedResources": {
+              "readOnly": true,
+              "type": "object",
+              "description": "The unused ports inbound details",
+              "properties": {
+                "unusedTcpPortsSummary": {
+                  "readOnly": true,
+                  "type": "integer",
+                  "description": "The unused inbound tcp ports summary"
+                },
+                "unusedUdpPortsSummary": {
+                  "readOnly": true,
+                  "type": "integer",
+                  "description": "The unused inbound udp ports summary"
+                },
+                "tcpPorts": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The unused inbound tcp ports"
+                },
+                "udpPorts": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The unused inbound udp ports"
+                }
+              }
+            },
+            "outboundConnectedResources": {
+              "readOnly": true,
+              "type": "object",
+              "description": "The unused ports outbound details",
+              "properties": {
+                "unusedTcpPortsSummary": {
+                  "readOnly": true,
+                  "type": "integer",
+                  "description": "The unused outbound tcp ports summary"
+                },
+                "unusedUdpPortsSummary": {
+                  "readOnly": true,
+                  "type": "integer",
+                  "description": "The unused outbound udp ports summary"
+                },
+                "tcpPorts": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The unused outbound tcp ports"
+                },
+                "udpPorts": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "The unused outbound udp ports"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "ConnectionDetails": {
+      "type": "object",
+      "description": "Details of the actual traffic connections",
+      "properties": {
+        "portAndProtocol": {
```

Comment:
why not in different fields?

Reply:
it will also allow you to use enum for the protocol

Reply:
we took the format from the NSG swagger.